lib/core/dsl.sh

```diff
@@ -313,7 +313,7 @@ shellspec_statement() {
 
 shellspec_when() {
   # Allow "When I"
-  if (( $# > 0 )) && [[ "$1" == "I" ]]; then
```

Comment:
This is not a problem with posh, it is [the specification of POSIX shell](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_01).